app/models/order_detail.rb

```diff
@@ -266,6 +265,11 @@ def can_be_viewed_by?(user)
       .where("dispute_at IS NULL OR dispute_resolved_at IS NOT NULL")
   }
 
+  scope :need_statement, lambda { |facility|
```

Comment:
I believe if you use `OrderDetail.for_facility(facility)`, it'll already handle the case of cross-facility.